packages/opentelemetry-node/src/spanprocessor.ts

```diff
@@ -111,5 +116,25 @@ function getTraceData(otelSpan: OtelSpan): Partial<TransactionContext> {
   return { spanId, traceId, parentSpanId };
 }
 
-// function updateSpanWithOtelData(sentrySpan: SentrySpan, otelSpan: OtelSpan): void {
-// }
+function updateContextWithOtelData(otelSpan: OtelSpan): void {
+  setContext('otel', {
```

Comment:
m: the alternate to this approach is to wrap `transaction.finish` with a `withScope` call that would set the context just while the transaction is being finished.

I’m in favour of doing that actually, because we use the existence of `otel` context to detect if an event comes from the processor, and we don’t want to “pollute” our errors with it.

Reply:
👍 pushed an update that should handle it this way. This required rewriting the tests a bit to take this into account - we now monkey-patch `transaction.finish()` to store the context at the time of finishing, so we can actually assert against that.